framework/src/actions/Action.C

```diff
@@ -37,9 +38,15 @@ InputParameters validParams<Action>()
 Action::Action(InputParameters parameters) :
     ConsoleStreamInterface(*parameters.getCheckedPointerParam<MooseApp *>("_moose_app", "In Action constructor")),
     _pars(parameters),
-    _name(getParam<std::string>("name")),
-    _short_name(MooseUtils::shortName(_name)),
     _registered_identifier(isParamValid("registered_identifier") ? getParam<std::string>("registered_identifier") : ""),
+    _full_syntax(getParam<std::string>("name")),
```

Comment:
This is to support action stuff and accessing parameters in parent blocks. Again, the auto traversal of the syntax would remove this.